ChangeLog

```diff
@@ -1,3 +1,35 @@
+3.12.1:
+	- Add commit title to changelog by default, if it has ticket number
+	  (ENT-3638)
+	- Class names set by module protocol are automatically canonified
+	  (CFE-2877, CFE-2887)
+	- Correct log level for data_readstringarray* (CFE-2922)
+	- Eliminated error messages caused by attempting to kill expired processes
+	  (CFE-2824)
+	- Extend the test to cover more cases (ENT-3959)
+	- Fix cf-runalerts systemd unit conditions so the service will run
+	  (ENT-3929)
+	- Fix the off-by-one error in cf-runagent background process spawning
+	  (CFE-2873)
+	- Fixed a memory leak which occurred when reloading RSA keys from disk
+	- Fixed a memory leak which occurred while loading augments files
+	  (CFE-2913)
+	- Fixed an issue with splay time in cf-execd (CFE-2931)
+	- Fixed error handling and memory leak in cf-key (CFE-2918)
+	- Fixed memory leak in lmdb cleanup (CFE-2918)
+	- Fixed memory leaks in cf-agent during bootstrap (CFE-2918)
+	- Fixed memory leaks in variablesmatching() and findfiles() (CFE-2918)
+	- Fixed missing class with mustache templates in warn_only mode (CFE-2600)
+	- Fixed small memory leak in cf-serverd (CFE-2918)
+	- Fixed valgrind warning about lmdb writing uninitialized memory (CFE-2591)
+	- Read mustache-rendered files in text mode when comparing digest (ENT-2526)
+	- Reload persistent classes on config reload in cf-execd and cf-serverd
+	  (CFE-2857)
+	- Test that classes can be defined without constraints (CFE-2000)
+	- Refactor use of atexit to use custom cleanup function instead. On Windows
+	  atexit() unloads DLLs before and/or during atexit functions being called
+	  which cases bad behavior. (ENT-3756)
```

Comment:
I would reword this one, not really a "refactor" per se. Something like: started using custom cleanup functions instead of `atexit()` which was causing race conditions related to DLL unloading on windows.